deployments.go

```diff
@@ -57,18 +59,21 @@ func (s *deployerService) deploy(ctx context.Context, db *gorm.DB, opts DeployOp
 		Config:      config,
 		Slug:        slug,
 		Description: desc,
-	})
+	}, ss)
 
 	return r, err
 }
 
 // Deploy is a thin wrapper around deploy to that adds the error to the
 // jsonmessage stream.
-func (s *deployerService) Deploy(ctx context.Context, db *gorm.DB, opts DeployOpts) (*Release, error) {
+func (s *deployerService) Deploy(ctx context.Context, opts DeployOpts) (*Release, error) {
 	var msg jsonmessage.JSONMessage
 
-	r, err := s.deploy(ctx, db, opts)
+	tx := s.db.Begin()
```

Comment:
I think we'd also wanna rollback here. Might be easier to wrap the transaction logic in a separate method:

```go
func (s *deployerService) deployInTransaction(ctx context.Context, stream scheduler.StatusStream, opts De
        tx := s.db.Begin()
        r, err := s.deploy(ctx, tx, stream, opts)
        if err != nil {
                tx.Rollback()
                return r, err
        }

        if err := tx.Commit().Error; err != nil {
                return r, err
        }

        return r, nil
}
```